cmd/hub/lifecycle/template_encodings.go

```diff
@@ -0,0 +1,140 @@
+package lifecycle
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/epam/hubctl/cmd/hub/util"
+	"gopkg.in/yaml.v2"
+)
+
+func isSecure(url *url.URL) bool {
+	return url.Scheme == "https"
+}
+
+func toURL(iface interface{}) (*url.URL, error) {
+	switch v := iface.(type) {
+	case string:
+		return parseURL(v)
+	case *url.URL:
+		return v, nil
+	default:
+		return nil, fmt.Errorf("invalid type %T", iface)
+	}
+}
+
+func parseURL(urlStr string) (*url.URL, error) {
+	u, err := url.Parse(urlStr)
+	if err != nil {
+		return nil, err
+	}
+	if u.Port() == "" {
+		if isSecure(u) {
+			u.Host = fmt.Sprintf("%s:443", u.Host)
+		} else if u.Scheme == "http" {
+			u.Host = fmt.Sprintf("%s:80", u.Host)
+		}
+	}
+	return u, nil
+}
+
+func encodingBase64(substitution interface{}) (interface{}, error) {
+	return base64.StdEncoding.EncodeToString([]byte(util.String(substitution))), nil
+}
+
+func encodingUnbase64(substitution interface{}) (interface{}, error) {
+	decoded, err := base64.StdEncoding.DecodeString(util.String(substitution))
+	if err != nil {
+		return nil, err
+	}
+	return string(decoded), nil
+}
+
+func encodingJson(substitution interface{}) (interface{}, error) {
+	jsonBytes, err := json.Marshal(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return string(jsonBytes), nil
+}
+
+func encodingYaml(substitution interface{}) (interface{}, error) {
+	yamlBytes, err := yaml.Marshal(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return string(yamlBytes), nil
+}
+
+func encodingFirst(substitution interface{}) (interface{}, error) {
+	str := util.String(substitution)
+	if strings.Contains(str, " ") {
+		return strings.Split(str, " ")[0], nil
+	}
+	return str, nil
+}
+
+func encodingParseUrl(substitution interface{}) (interface{}, error) {
+	str := util.String(substitution)
+	url, err := parseURL(str)
+	if err != nil {
+		return nil, err
+	}
+	return url, nil
+}
+
+func encodingIsUrlSecure(substitution interface{}) (interface{}, error) {
+	url, err := toURL(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return isSecure(url), nil
+}
+
+func encodingIsUrlInsecure(substitution interface{}) (interface{}, error) {
+	url, err := toURL(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return !isSecure(url), nil
+}
+func encodingGetUrlHostname(substitution interface{}) (interface{}, error) {
+	url, err := toURL(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return url.Hostname(), nil
+}
+func encodingGetUrlPort(substitution interface{}) (interface{}, error) {
+	url, err := toURL(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return url.Port(), nil
+}
+func encodingGetUrlScheme(substitution interface{}) (interface{}, error) {
+	url, err := toURL(substitution)
+	if err != nil {
+		return nil, err
+	}
+	return url.Scheme, nil
+}
+
+var encodingsMap = map[string]func(interface{}) (interface{}, error){
```

Comment:
Looks like variable name doesn't match intention anymore. Now the variable contains `parseURL` which is not an encoding by any stretch. Also the corresponding message at call site.
Suggest `converter` or just `function`.